src/IRTS/CodegenC.hs

```diff
@@ -179,8 +179,10 @@ bcc i (ASSIGNCONST l c)
     = indent i ++ creg l ++ " = " ++ mkConst c ++ ";\n"
   where
     mkConst (I i) = "MKINT(" ++ show i ++ ")"
-    mkConst (BI i) | i > ((-2)^30) && i < (2^30) = "MKINT(" ++ show i ++ ")"
-                   | otherwise = "MKBIGC(vm,\"" ++ show i ++ "\")"
+    mkConst (BI i) = let maxInt = 2^30
+                     in if i >= -maxInt && i < maxInt
```

Comment:
You could use `if i < abs 2^30` for simplicity

Reply:
You mean `abs i < abs 2^30`

Reply:
The test is different on either side of the interval.

Reply:
No I put the abs on the wrong side, 2^30 don't need an abs, it's a constant :-)

Reply:
or `abs i < 2^30`

Reply:
spoiler alert: `x^30 = abs (x^30)` for any real `x` :D